lib/devise/models/confirmable.rb

```diff
@@ -121,8 +125,18 @@ def send_reconfirmation_instructions
         @reconfirmation_required = false
 
         unless @skip_confirmation_notification
-          send_confirmation_instructions
```

Comment:
we should still can `send_confirmation_instructions` if `Devise.send_reconfirmation_instructions` is false, otherwise we will break existing apps.